.github/workflows/check_dependencies.yml

```diff
@@ -0,0 +1,41 @@
+name: Check for dependency modification
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```

Comment:
Is there any reason that we should run this workflow `on pull_request`?

Reply:
This was due to the overengineering part. We definitely could just trigger it on a push to main.